packages/wrangler/e2e/dev.test.ts

```diff
@@ -117,6 +117,57 @@ describe("basic dev tests", () => {
 		});
 	});
 
+	it("can recover from syntax error during dev session (local)", async () => {
+		await runDevSession(workerPath, "", async (port) => {
+			const { text } = await retry(
+				(s) => s.status !== 200,
+				async () => {
+					const r = await fetch(`http://127.0.0.1:${port}`);
+					return { text: await r.text(), status: r.status };
+				}
+			);
+			expect(text).toMatchInlineSnapshot('"Hello World!"');
+
+			await seed(workerPath, {
+				"src/index.ts": dedent`
+						export default {
+							fetch(request, env) {
+								return new Response("Updated Worker!")
+								} // <= added syntax error
+							}
+						}`,
+			});
+
+			// Output during error state is indeterminate.
+			// const { text: text2 } = await retry(
+			// 	(s) => s.status !== 200 || s.text === "Hello World!",
+			// 	async () => {
+			// 		const r = await fetch(`http://127.0.0.1:${port}`);
+			// 		return { text: await r.text(), status: r.status };
+			// 	}
+			// );
+			// expect(text2).toMatchInlineSnapshot('"Hello World!"');
```

Comment:
For this test to be worth running, we need to confirm the syntax error took effect. Without these steps, the seed function is called immediately after the last and so could lead to a race condition as to whether wrangler/miniflare/workerd even saw the syntax error

Can you describe what you mean by "indeterminate"? Maybe just asserting the response is _not_ "Hello World" is enough?